crates/sui-core/src/authority_server.rs

```diff
@@ -309,6 +309,20 @@ impl ValidatorService {
             }
             .into());
         }
+        println!("transaction: {:?}", transaction);
+        println!(
+            "protocol_config: {:?}",
+            epoch_store.protocol_config().supports_upgraded_multisig()
+        );
+        println!("1: {:?}", transaction.has_upgraded_multisig());
+        if !epoch_store.protocol_config().supports_upgraded_multisig()
+            && transaction.has_upgraded_multisig()
+        {
+            return Err(SuiError::UnsupportedFeatureError {
+                error: "upgraded multisig format not enabled on this network".to_string(),
+            }
+            .into());
+        }
```

Comment:
this test should be done in `impl VersionedProtocolMessage for SenderSignedData` in transaction.rs - that function already iterates over all signatures to see if they are supported.

Reply:
you should be able to delete this code now, the check in transaction.rs should be enough